core/src/components/menu/menu.tsx

```diff
@@ -169,26 +169,11 @@ export class Menu implements ComponentInterface, MenuI {
       return;
     }
 
-    const el = this.el;
-    const parent = el.parentNode as any;
-
-    if (this.contentId === undefined) {
-      console.warn(`[DEPRECATED][ion-menu] Using the [main] attribute is deprecated, please use the "contentId" property instead:
-BEFORE:
-  <ion-menu>...</ion-menu>
-  <div main>...</div>
-
-AFTER:
-  <ion-menu contentId="main-content"></ion-menu>
-  <div id="main-content">...</div>
-`);
-    }
     const content = this.contentId !== undefined
       ? document.getElementById(this.contentId)
-      : parent && parent.querySelector && parent.querySelector('[main]');
+      : null;
 
-    if (!content || !content.tagName) {
-      // requires content element
+    if (content === null) {
```

Comment:
If `content` is defined as an element, then `tagName` is defined: https://developer.mozilla.org/en-US/docs/Web/API/Element/tagName. Nothing on MDN indicates tagName could be the empty string (which is falsy in this case)